source/modulo_components/test/cpp/test_component_interface.cpp

```diff
@@ -7,59 +7,73 @@
 #include "modulo_new_core/EncodedState.hpp"
 
 namespace modulo_components {
-class TestComponentInterface : public ::testing::Test {
+class ComponentInterfaceTest : public ::testing::Test {
 protected:
   static void SetUpTestSuite() {
     rclcpp::init(0, nullptr);
-    if (component_ == nullptr) {
-      component_ =
-          new ComponentInterface<rclcpp::Node, rclcpp::Publisher<modulo_new_core::EncodedState>>(rclcpp::NodeOptions());
-    }
   }
 
-  void TearDown() override {
-    component_->predicates_.clear();
+  void SetUp() override {
+    component_ = std::make_unique<ComponentInterface<rclcpp::Node, rclcpp::GenericPublisher>>(rclcpp::NodeOptions());
   }
 
-  static ComponentInterface<rclcpp::Node, rclcpp::Publisher<modulo_new_core::EncodedState>>* component_;
-};
+  std::map<std::string, utilities::PredicateVariant> get_predicate_map() {
+    return component_->predicates_;
+  }
+
+  void add_predicate(const std::string& name, bool value) {
+    component_->add_predicate(name, value);
+  }
+
+  void add_predicate(const std::string& predicate_name, const std::function<bool(void)>& predicate_function) {
+    component_->add_predicate(predicate_name, predicate_function);
+  }
 
-ComponentInterface<rclcpp::Node, rclcpp::Publisher<modulo_new_core::EncodedState>>
-    * TestComponentInterface::component_ = nullptr;
+  bool get_predicate(const std::string& predicate_name) const {
+    return component_->get_predicate(predicate_name);
+  }
+
+  void set_predicate(const std::string& predicate_name, bool predicate_value) {
+    component_->set_predicate(predicate_name, predicate_value);
+  }
+
+  std::unique_ptr<ComponentInterface<rclcpp::Node, rclcpp::GenericPublisher>> component_;
```

Comment:
Because we inherit from `std::enable_shared_from_this`, it really is necessary for the component interface to be held by a shared pointer and not a unique pointer. Otherwise, we'll have `std::bad_weak_ptr` or undefined exceptions at a later point. Unique pointers should not be used to manage nodes.